src/Microsoft.ML.Transforms/GcnTransform.cs

```diff
@@ -760,8 +658,85 @@ public enum NormalizerKind : byte
+            /// </summary>
+            public readonly NormalizerKind NormKind;
+            /// <summary>
+            /// Scale features by this value.
+            /// </summary>
+            public readonly float Scale;
+
+            internal ColumnInfoBase(string name, string inputColumnName, bool substractMean, NormalizerKind normalizerKind, float scale)
+            {
+                Contracts.CheckNonWhiteSpace(name, nameof(name));
+                Contracts.CheckNonWhiteSpace(inputColumnName, nameof(inputColumnName));
+                Name = name;
+                InputColumnName = inputColumnName;
+                SubtractMean = substractMean;
+                Contracts.CheckUserArg(0 < scale && scale < float.PositiveInfinity, nameof(scale), "scale must be a positive finite value");
+                Scale = scale;
+                NormKind = normalizerKind;
+            }
+
+            internal ColumnInfoBase(ModelLoadContext ctx, string name, string inputColumnName, bool normKindSerialized)
+            {
+                Contracts.AssertValue(ctx);
+                Contracts.CheckNonWhiteSpace(inputColumnName, nameof(inputColumnName));
+                Contracts.CheckNonWhiteSpace(name, nameof(name));
+                Name = name;
+                InputColumnName = inputColumnName;
+
+                // *** Binary format ***
+                // byte: SubtractMean
+                // byte: NormKind
+                // Float: Scale
+                SubtractMean = ctx.Reader.ReadBoolByte();
+                byte normKindVal = ctx.Reader.ReadByte();
+                Contracts.CheckDecode(Enum.IsDefined(typeof(NormalizerKind), normKindVal));
+                NormKind = (NormalizerKind)normKindVal;
+                // Note: In early versions, a bool option (useStd) to whether to normalize by StdDev rather than
+                // L2 norm was used. normKind was added in version=verVectorNormalizerSupported.
+                // normKind was defined in a way such that the serialized boolean (0: use StdDev, 1: use L2) is
+                // still valid.
+                Contracts.CheckDecode(normKindSerialized ||
+                        (NormKind == NormalizerKind.L2Norm || NormKind == NormalizerKind.StdDev));
+                Scale = ctx.Reader.ReadFloat();
+                Contracts.CheckDecode(0 < Scale && Scale < float.PositiveInfinity);
+            }
+
+            internal void Save(ModelSaveContext ctx)
+            {
+                Contracts.AssertValue(ctx);
+                // *** Binary format ***
+                // byte: SubtractMean
+                // byte: NormKind
+                // Float: Scale
+                ctx.Writer.WriteBoolByte(SubtractMean);
+                ctx.Writer.Write((byte)NormKind);
+                Contracts.Assert(0 < Scale && Scale < float.PositiveInfinity);
+                ctx.Writer.Write(Scale);
+            }
+        }
+
+        public static class Defaults
```

Comment:
>public [](start = 8, length = 6)

why public?

Reply:
Thank you!

---
In reply to: [254045932](https://github.com/dotnet/machinelearning/pull/2366#discussion_r254045932) [](ancestors = 254045932)